app/shared/productTypes.tsx

```diff
@@ -1,25 +1,43 @@
 import React from "react";
-import { NoMembership } from "../client/components/cancel/membership/noMembership";
-import { MeValidator } from "../client/components/cancellationFlowWrapper";
+import { MeValidator } from "../client/components/cancel/cancellationFlowWrapper";
+import { NoMembership } from "../client/components/noMembership";
 import { MeResponse } from "./meResponse";
 
-export type ProductName = "membership" | "recurring contribution";
+export type ProductFriendlyName = "membership" | "recurring contribution";
+export type ProductUrlPart = "membership" | "contributions";
 
 export interface ProductType {
-  productName: ProductName;
+  friendlyName: ProductFriendlyName;
+  urlPart: ProductUrlPart;
   validator: MeValidator;
   invalidComponentRenderer: JSX.Element;
+  fetchProductDetail: () => Promise<Response>;
+  tierRowLabel?: string; // no label means row is not displayed
 }
 
+const createProductFetcher = (productApiKeyword: string) => async () =>
```

Comment:
Could you make the type of this argument a ProductUrlPart?

Reply:
i've actually reworked this in subsequent PR (https://github.com/guardian/manage-frontend/pull/129)